platform-hub-api/app/models/kubernetes_token.rb

```diff
@@ -69,23 +71,25 @@ def obfuscated_token
     val[0..30].gsub(/\w/, 'X') + val[31..35]
   end
 
-  def owner
-    if self.user?
-      case self.tokenable
-      when Identity
-        self.tokenable.user
-      end
-    end
-  end
-
   def privileged?
     expire_privileged_at.present?
   end
 
-  def escalate(privileged_group, expire_in_secs = 600)
-    update_attributes(
+  def escalate(privileged_group_name, expire_in_secs = 600)
+    group = KubernetesGroup.where(name: privileged_group_name).first
+
+    if group.blank?
+      raise ActiveRecord::RecordNotFound
```

Comment:
if you use `.find_by!` above you can skip this check.

Reply:
Good idea!